### ISSUE TITLE
API structure

### ISSUE THREAD

**Issue by theogf** [theogf/BayesianQuadrature.jl#1]
So Bayesian quadrature is quite unlike other quadrature packages, therefore we need to think how to approach it.

On top of the likelihood/prior input, there are two important playing parameters:
- The method to select new samples
- The method to evaluate the different integrals (vanilla or with transformation)

I think for simplicity it is better to pass these two separately as there is no real benefit to embed the first one in the second.

**Comment by theogf** [comment]
An additional point I forgot to consider was on how to compute the kernel means, i.e :
m(x) = \int k(x, x') p(x')dx'

**Comment by johannesgiersdorf** [comment]
> On top of the likelihood/prior input, there are two important playing parameters:
> 
>     * The method to select new samples
We should keep in mind that some methods depend on prev. produced samples (e.g. active sampling) and it should be easy to create samples in batches (for hyper parameter estimation).

**Comment by theogf** [comment]
Yep that's the idea! Have a look at AbstractMCMC.jl I think we should use their interface.
